config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -280,6 +284,41 @@ public ServerHttpSecurity securityContextRepository(ServerSecurityContextReposit
 		return this;
 	}
 
+	/**
+	 * Configures HTTPS redirection rules. If the default is used:
+	 *
+	 * <pre class="code">
+	 *  &#064;Bean
+	 * 	public SecurityWebFilterChain springSecurityFilterChain(ServerHttpSecurity http) {
+	 * 	    http
+	 * 	        // ...
+	 * 	        .redirectToHttps();
+	 * 	    return http.build();
+	 * 	}
+	 * </pre>
+	 *
+	 * Then all non-HTTPS requests will be redirected to HTTPS.
+	 *
+	 * Typically, all requests should be HTTPS; however, the focus for redirection can also be narrowed:
```

Comment:
Please change this example to be something that we might recommend doing. For example, if a production header is in the request then redirect to HTTPS. This makes it simple to enable HTTPS for production but not other environments that don't have HTTPS support.